invenio_cli/helpers/package_managers.py

```diff
@@ -0,0 +1,173 @@
+        package_version = package if not version else package + version
+        return [self.name, "install", package_version]
+
+    def update_packages(self):
+        """Update all updated packages."""
+        return [self.name, "update"]
+
+    def list_outdated_packages(self):
+        """List outdated installed packages."""
+        return [self.name, "update", "--outdated"]
+
+    def install_locked_deps(self, prereleases, devtools):
+        """Install the packages according to the lock file."""
+        cmd = [self.name, "sync"]
+        if prereleases:
+            cmd += ["--pre"]
+        if devtools:
+            cmd += ["--dev"]
+        return cmd
+
+    def lock_dependencies(self, prereleases, devtools):
+        """Update the lock file."""
+        cmd = [self.name, "lock"]
+        if prereleases:
+            cmd += ["--pre"]
+        if devtools:
+            cmd += ["--dev"]
+        return cmd
+
+    def remove_venv(self):
+        """Remove the created virtualenv."""
+        return ["pipenv", "--rm"]
+
+    def start_activated_subshell(self) -> List[str]:
+        """Remove the created virtualenv."""
+        return ["pipenv", "shell"]
+
+
+class UV(PythonPackageManager):
+    """Generate ``uv`` commands for managing Python packages."""
+
+    name = "uv"
+    lock_file_name = "uv.lock"
+
+    def run_command(self, *command):
+        """Generate command to run the given command in the managed environment."""
+        # "--no-sync" is used to not override locally installed editable packages
+        return [self.name, "run", "--no-sync", *command]
+
+    def editable_dev_install(self, *packages):
+        """Install the local packages as editable, but ignore it for locking."""
+        cmd = [self.name, "pip", "install"]
+        for package in packages:
+            cmd += ["-e", package]
+        return cmd
+
+    def install_package(self, package, version=None):
+        """Install the package in the specified version."""
+        package_version = package if not version else package + version
+        return [self.name, "add", package_version]
+
+    def update_packages(self):
+        """Update all updated packages."""
+        return [self.name, "sync", "--upgrade"]
+
+    def list_outdated_packages(self):
+        """List outdated installed packages."""
+        return [self.name, "sync", "--upgrade", "--dry-run"]
+
+    def install_locked_deps(self, prereleases, devtools):
+        """Install the packages according to the lock file."""
+        cmd = [self.name, "sync"]
+        if prereleases:
+            cmd += ["--prerelease", "allow"]
+        if not devtools:
+            cmd += ["--no-dev"]
+        return cmd
+
+    def lock_dependencies(self, prereleases, devtools):
+        """Update the lock file."""
+        cmd = [self.name, "lock"]
+        if prereleases:
+            cmd += ["--prerelease", "allow"]
+        return cmd
+
+    def remove_venv(self):
```

Comment:
i couldn't find `uv` commands for those; if you know more than me here, please let me know then we can update this.

Reply:
I guess we explicitly use `.venv` instead of getting it from config or similar? I'm not seeing that `pipenv --rm` is better in that manner, since it will just try to deduce based on its usual rules where the virtualenv is, e.g. `$HOME/Envs`, `.venv`, currently activated virtualenv (?), etc.

Reply:
i've looked for a way to make `uv` tell me where it's hiding its venv, but it wouldn't say.
apparently it only prints the location on creation.
until then i guess we'll just assume it's under the floor boards.